provider/pkg/provider/provider.go

```diff
@@ -329,24 +329,39 @@ func (k *kubeProvider) DiffConfig(ctx context.Context, req *pulumirpc.DiffReques
 	if !reflect.DeepEqual(oldConfig, newConfig) {
 		diffs = append(diffs, "kubeconfig")
 	}
-	if olds["context"] != news["context"] {
-		diffs = append(diffs, "context")
-	}
 	if olds["cluster"] != news["cluster"] {
 		diffs = append(diffs, "cluster")
 	}
-	if olds["namespace"] != news["namespace"] {
-		diffs = append(diffs, "namespace")
+	if olds["context"] != news["context"] {
+		diffs = append(diffs, "context")
 	}
 	if olds["enableDryRun"] != news["enableDryRun"] {
 		diffs = append(diffs, "enableDryRun")
 	}
+	if olds["enabledReplaceCRD"] != news["enableReplaceCRD"] {
+		diffs = append(diffs, "enableReplaceCRD")
+	}
+	if olds["helmReleaseSettings"] != news["helmReleaseSettings"] {
+		diffs = append(diffs, "helmReleaseSettings")
+	}
+	if olds["kubeClientSettings"] != news["kubeClientSettings"] {
+		diffs = append(diffs, "kubeClientSettings")
+	}
+	if olds["namespace"] != news["namespace"] {
+		diffs = append(diffs, "namespace")
+	}
 	if olds["renderYamlToDirectory"] != news["renderYamlToDirectory"] {
 		diffs = append(diffs, "renderYamlToDirectory")
 
-		// If the render directory changes, all of the manifests will be replaced.
+		// If the render directory changes, all the manifests will be replaced.
 		replaces = append(replaces, "renderYamlToDirectory")
 	}
+	if olds["suppressDeprecationWarnings"] != news["suppressDeprecationWarnings"] {
+		diffs = append(diffs, "suppressDeprecationWarnings")
+	}
+	if olds["suppressHelmHookWarnings"] != news["suppressHelmHookWarnings"] {
+		diffs = append(diffs, "suppressHelmHookWarnings")
+	}
```

Comment:
Can we introspect on the schema and look through the config settings instead of hardcoding/checking these? Seems like that would avoid the whole possibility of falling out of sync here?

Reply:
Yeah, I think another option would be iterating the entire struct, and special-casing any we need to handle differently. That would at least give a default diff.

Reply:
Let me know how you want to proceed. Do the above now or merge this and follow-up. I do worry about the drift so the more we can standardize around schema guiding operations the better. This would also be a great endorsement for common logic living in a library across providers.

Reply:
I switched to iterating the keys. Let's figure out how to centralize this logic as a follow up.